components/calendar/src/date.rs

```diff
@@ -129,21 +130,36 @@ impl<A: AsCalendar> Date<A> {
     ) -> Result<Self, DateError> {
         let inner = calendar
             .as_calendar()
-            .date_from_codes(era, year, month_code, day)?;
+            .from_codes(era, year, month_code, day)?;
         Ok(Date { inner, calendar })
     }
 
+    /// Construct a date from a [`RataDie`] and some calendar representation
+    #[inline]
+    pub fn from_rata_die(rd: RataDie, calendar: A) -> Self {
+        Date {
+            inner: calendar.as_calendar().from_rata_die(rd),
+            calendar,
+        }
+    }
```

Comment:
Question/Issue: This is not proposed as a fallible function. Currently all of our functions are fallible, which means they can return out-of-range errors. We've had the _convention_ that all our calendars should be constructible for all dates in the RataDie range, but it's only been an internal convention, and not really something where we've had an official policy. Are we sure that we want to expose this in our public API?